module/zfs/zfs_vfsops.c

```diff
@@ -1701,8 +1701,7 @@ zfs_resume_fs(zfs_sb_t *zsb, const char *osname)
 		 * Since we couldn't setup the sa framework, try to force
 		 * unmount this file system.
 		 */
-		if (zsb->z_os)
-			(void) zfs_umount(zsb->z_sb);
+		(void) zfs_umount(zsb->z_sb);
```

Comment:
The original fix here doesn't look quite right.  Take a look at the stack from the original bug report.  If zsb->z_os were NULL we should never have been able to make it to where the node finally crashed.

https://github.com/zfsonlinux/zfs/issues/1518

```
zfs_resume_fs
  zfs_umount
    zfs_sb_teardown
      dmu_objset_pool <- PANIC
```

In `zfs_sb_teardown` several checks for `zsb->z_os == NULL` are made before it's used which should have prevented this.  What exactly in this first check preventing that the other ones aren't?